churn-risk/src/churn_predictor.py

```diff
@@ -11,85 +12,57 @@ class ChurnPredictor:
-        )
-
-    def get_shap_explanation(self, row_index):
-        return self.model.shap_explain_row_plot(frame=self.test_df, row_index=row_index)
-
-    def get_top_negative_pd_explanation(self, row_index):
-        """
-        Return the partial dependence explanation of the top negatively contributing feature.
-
-        :param row_index: row index to select from H2OFrame for the explanation
-        :return: matplotlib figure object
-        """
-        # column_index = self.contributions_df.idxmin(axis=1).as_data_frame()[
-        #     "which.min"
-        # ][row_index]
-
-        # Using Pandas DataFrame.idxmin() until https://h2oai.atlassian.net/browse/PUBDEV-7906 is fixed
-        pdf = self.contributions_df.as_data_frame()
-        del pdf["BiasTerm"]  # Delete bias column added by predict_contributions()
-        min_column_name = pdf.idxmin(axis=1)[row_index]
-
-        return self.model.pd_plot(
-            frame=self.test_df,
-            row_index=row_index,
-            column=min_column_name,
-        )
-
-    def get_top_positive_pd_explanation(self, row_index):
-        """
-        Return the partial dependence explanation of the top positively contributing feature.
-
-        :param row_index: row index to select from H2OFrame for the explanation
-        :return: matplotlib figure object
-        """
-        column_index = self.contributions_df.idxmax(axis=1).as_data_frame()[
-            "which.max"
-        ][row_index]
-        return self.model.pd_plot(
-            frame=self.test_df,
-            row_index=row_index,
-            column=self.test_df.col_names[column_index],
-        )
+        # Initialize H2O-3 model and tests data set
+        self.train_df = h2o.import_file(path='./data/churnTrain.csv', destination_frame="telco_churn_train.csv")
+
+        train, valid = self.train_df.split_frame([0.8])
+        self.model = H2OGradientBoostingEstimator(model_id="telco_churn_model", seed=1234)
+        self.model.train(x=self.train_df.columns, y="Churn?", training_frame=train, validation_frame=valid)
+
+        self.h2o_test_df = h2o.import_file(path='./data/churnTest.csv', destination_frame="telco_churn_test.csv")
+        self.predicted_df = self.model.predict(self.h2o_test_df).as_data_frame()
+        self.contributions_df = self.model.predict_contributions(self.h2o_test_df).drop('BiasTerm').as_data_frame()
+
+    def get_churn_rate(self, row_index: int):
+        return round(float(self.predicted_df["TRUE"][row_index]) * 100, 2)
+
+    def get_shap(self, row_index: int):
+        np_row = self.contributions_df.iloc[row_index].to_numpy()
+        shap= [(self.contributions_df.columns[i], np_row[i]) for i in range(len(self.contributions_df.columns))]
+        shap.sort(key=lambda e : e[1])
+        return shap 
+    
+    def get_python_type(self, val):
+        return val if isinstance(val, (str, float)) else val.item()
+        
+    def get_size(self, group_size, idx: int):
```

Comment:
This can be private and a class method if `_get_python_type()` made the way I described above:

```python3
@classmethod
def _get_size(cls, group_size, idx: int):
    return 0 if idx > len(group_size) - 1 else cls._get_python_type(group_size[idx])
```

Reply:
Will do, thanks